Comments from Mark Taylor added

aibaiba opened the pull request.

Commit:
Updated with comments from Mark Taylor

Commit:
New WD date

gmantele reviewed (dismissed):
This looks OK to me. But let's wait for Mark review before merging as he is the comments author.

mbtaylor committed:
fix example VOTable

Note that the "../v1.3" is correct in the xmlns attribute, even
though the VOTable version is 1.5.

mbtaylor reviewed (changes requested):
I made a validity fix to the example VOTable in section 5.

Other than that it looks good with one exception: the `t_validity` field is documented as mandatory, but it does not appear in the example response in section 5.  Either downgrade `t_validity` to optional or add a suitable column to the VOTable.

esabol commented on ObjObsSAP.tex:
```diff
@@ -554,100 +527,95 @@ \subsection{Successful Query}
-confidence of the validity range, with one of the following allowed
+\begin{itemize}
+\item {A numeric field with name="\textbf{t\_validity}" \textbf{MUST} be present  
+with utype=" \textbf{Char.TimeAxis.Coverage.Time}" and unit="d", containing the date when the 
```
For the second time, I am submitting the same review comment: Please remove the space after `utype="`.

aibaiba replied:
Done

jwfraustro commented on ObjObsSAP.tex:
```diff
@@ -278,40 +279,32 @@ \subsubsection{Required parameters}
-the client to limit the number or records in the response. A service
+the client to limit the number of records in the response. A service
 implementation may also impose default and maximum values for this limit.
 However, the limit is determined, if the output is truncated due to the
```
```suggestion
However the limit is determined, if the output is truncated due to the
```

Since this is a subordinate clause, the comma should appear after the clause and not the subordinating conjunction.

jwfraustro commented on ObjObsSAP.tex:
```diff
@@ -278,40 +279,32 @@ \subsubsection{Required parameters}
 \textbf{TIME} parameter, to specify the time coverage in range-list form
 to check for object observability. The unit of \textbf{TIME} parameter must be
 expressed in MJD. The format of the range list is the one defined by other
-IVOA S*APs protocols like SSAP \citep{std:SSAP}, where the format of a
+IVOA S*APs protocols like SSAP \citep{2012ivoa.spec.0210T}, where the format of a
 range is defined by '$t_{min}\slash t_{max}$'. If the range is defined
-as an open range without the lower value of the range like '$\slash t_{max}$',
+as an open range without the lower value of the range like '$t_{max}$',
 $t_{min}$ will be interpreted as now.\\
```
I don't think the reference to the other Simple protocols is entirely correct or, at least, may be confusing to the reader.

For `TIME`: 
- SSA does indicate its range-list values this way, with a `/`, but uses ISO-8601. 
- SIA's `TIME` does use MJD, but indicates its range-lists with a space.

Maybe it's best to drop the reference entirely? Either place the reader looks, the examples would be incorrect.

aibaiba replied:
Fromt the SSA doc: 
Allowable ISO8601 formats include the date (e.g., yyyy-mm-dd with the month
and day fields being optional, the minimum being yyyy), or the date-time (e.g., yyyy-mmddThh:mm:ss). 

I will leave it as it is

gmantele replied:
I would tend to say that we should not reference SSA. SIA and SSA should be merged at some point into DAP. That's why, referencing SSA (or SIA) does not seem like a good idea. Anyway, now, we have DALI in DAL which should be the basis of all DAL standards (SSA and SIA came after the birth of DALI). So, for new DAL standards, the recommendation is to follow as much as possible DALI. That what will happen to ConeSearch-1.1, for instance. So, in ObjObsSAP, I think that we should generally refer to DALI, when possible.

In this particular case, DALI-1.1 and 1.2 explicitly recommend to use an ISO8601 (like) format. There is an xtype for range or interval, but none applies to timestamps (but instead to numeric values). Maybe that should be something to speak about with Pat.

jwfraustro replied:
> Fromt the SSA doc: Allowable ISO8601 formats include the date (e.g., yyyy-mm-dd with the month and day fields being optional, the minimum being yyyy), or the date-time (e.g., yyyy-mmddThh:mm:ss).
> 
> I will leave it as it is

I agree with Grégory's point overall that DALI should be followed, so my correction is somewhat moot.

The point of my original comment was to point out that these are the ways of providing a time range in ObjObsSAP and the above mentioned protocols:

SSA: `2025-01-01/2025-12-30` <- ISO(SLASH)ISO
SIA: `59522 59532` <- MJD(SPACE)MJD
ObjObsSAP: `59522/59532` <- MJD(SLASH)MJD

So the statement I suggested correcting:
> The format of the range list is the one defined by other IVOA S*APs protocols like SSAP [], where the format of a range is defined by 'tmin/tmax'

is because ObjObsSAP _does not_ specify its time-ranges like the other S*AP protocols (since they don't agree with each other anyway)

esabol replied:
I also agree with following DALI. Both SIA and SSA are older (some might say outdated even) standards, which need to be updated, imho, and do not reflect current IVOA preferred usage..

jwfraustro commented on ObjObsSAP.tex:
Since `RESPONSEFORMAT` is implied as being supported in `Sec 4.1`, perhaps it should be added to the `Non-compulsory Parameters` section?

jwfraustro commented on ObjObsSAP.tex:
```diff
@@ -808,38 +770,38 @@ \section{ObjObsSAP data model summary}
-\textbf{moon\_sep\_min} & \textbf{(MAY)} & phys.angDist & Minimum
+\textbf{moon\_sep\_min} & \textbf{(MAY)} &  & Minimum
 Moon separation for this sky position and observability time interval &
 float & deg \\
```
It looks like `moon_sep_min` was missed as a `numeric`.
```suggestion
numeric & deg \\
```

jwfraustro commented on ObjObsSAP.tex:
```diff
@@ -798,7 +760,7 @@ \section{ObjObsSAP data model summary}
 (MAY)} &
 em.energy & Energy minimum for this particular sky position and
-observability time interval & float & m \\
+observability time interval & numeric & m \\
```
Should `em_max` be a numeric as well? See `moon_sep_min` comment.

aibaiba replied:
Done

aibaiba committed:
Apply suggestions from code review

Co-authored-by: Joshua Fraustro <36318163+jwfraustro@users.noreply.github.com>

aibaiba committed:
Update ObjObsSAP.tex

Co-authored-by: Joshua Fraustro <36318163+jwfraustro@users.noreply.github.com>

aibaiba committed:
fix the utype white space and the em_max numeric type

aibaiba reviewed (commented):
fix issues from esabol and jwfraustro

aibaiba committed:
Merge branch 'master' into aitor-patch-2

mbtaylor reviewed (approved):
Thanks @aibaba.  This looks OK except for the unaddressed comment that I made on 30 May:
> the t_validity field is documented as mandatory, but it does not appear in the example response in section 5. Either downgrade t_validity to optional or add a suitable column to the VOTable.

I approve this PR; if you want to merge it without addressing the above it's fine, I can report that as a new issue.

esabol commented on ObjObsSAP.tex:
```diff
@@ -563,100 +536,95 @@ \subsection{Successful Query}
-\item {Exactly one field \textbf{MAY} have a name="\textbf{validity\_accuracy}" 
-with datatype="char" and arraysize="*" containing the level of
-confidence of the validity range, with one of the following allowed
+The list of the output fields are descrined below.
```
Typo here: `descrined` should be `described`
